src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,121 @@
+// eslint-disable-next-line object-curly-newline
+import classNames from 'classnames';
+import { useEffect, useMemo, useRef } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import { Todo } from '../../types/Todo';
+import { Header } from '../Header';
+import { TodoList } from '../TodoList';
+import { TodosFilter } from '../TodosFilter';
+import { filterTotos } from '../../helpers/filterTotos';
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+
+  const completedTodosCount = useMemo(
+    () => todos.filter((todo: Todo) => todo.completed).length,
+    [todos],
+  );
+  const activeTodosCount = useMemo(
+    () => todos.filter((todo: Todo) => !todo.completed).length,
+    [todos],
+  );
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const allCompleted = completedTodosCount === todos.length;
+  const { pathname: location } = useLocation();
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, []);
+
+  const handleAddTodo = (text: string) => {
+    const newTodo: Todo = {
+      id: +new Date(),
+      title: text,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
Replace with `setTodos((todosList) => [...todosList, newTodo]);`

Reply:
fix it
![image](https://user-images.githubusercontent.com/98764641/222369883-b5119ae8-ce78-4c85-b708-3b802324df0f.png)

Reply:
Not resolved. If you do something like this: `setTodos([...todos, newTodo]);`
problems may arise if you have to change the state several times in one function.

Bad code example:

```
setState([...state, "A"])
setState([...state, "B"])
```

`state` will be `[...., "B"]`


Good code example:

```
setState((state) => [...state, "A"])
setState((state) => [...state, "B"])
```

`state` will be `[...., "A", "B"]`